packages/crashlytics/ios/RNFBCrashlytics/RNFBCrashlyticsInitProvider.m

```diff
@@ -79,26 +79,13 @@ + (BOOL)isCrashlyticsJavascriptExceptionHandlerChainingEnabled {
 }
 
 + (NSArray<FIRComponent *> *)componentsToRegister {
-  return @[];
-}
-
-/*
- * configureWithApp is automatically invoked by Firebase as this app is a registered FIRLibrary with
- * the SDK.
- *
- * At this point "configure" has already been called on the FIRApp instance. This behavior is meant
- * to mirror what is done in ReactNativeFirebaseCrashlyticsInitProvider.java
- *
- * This pattern may not be supported long term
- * https://github.com/firebase/firebase-ios-sdk/issues/2933
- */
-+ (void)configureWithApp:(FIRApp *)app {
```

Comment:
Thanks, @mikehardy! I need to investigate a little further but one potential issue with this is that, as I understand it, `componentToRegister` runs before `configureWithApp:`, so it may not work to move the Crashlytics bit into `componentToRegister` as the Crashlytics instance may not be ready. Were you able to run this and see if `[FIRCrashlytics crashlytics]` returns a non-nil instance?

Reply:
Well, it ran and did not have a nil dereference as the code accessed the instance during any test runs .. perhaps it instantiates crashlytics itself? Certainly don't want to rely on winning a race condition though, happy for any guidance

Reply:
@ncooke3 okay okay :-), in lieue of the "easy cheesy" solution (which frankly gives me anxiety since you've mentioned it could be racy and have a nil crashlytics instance) I have attempted to use the full component infrastructure to create a dependency on crashlytics, then ask the component system to instantiate our provider here eagerly for the default app, using the dependency

It appears to work in testing the same as the previous "just run it" style did, but there should be no way this can get a nil crashlytics if I understand things correctly?

Would love a set of eyes on it though if you have a chance

Reply:
Hi @mikehardy, the one issue now is that `FIRDependency` is also staged to go away. The In App Messaging iOS SDK uses `configureWithApp:` so I've been thinking through if/how we could generalize a replacement solution for both. 

Where does the `self.isCrashlyticsCollectionEnabled` value come from?

Reply:
Okay, your latest implementation ended up needed only a few tweaks and it should be gtg. Leaving some review comments...

Reply:
Invaluable review pushing me to a more reliable implementation and then polishing that, really appreciated @ncooke3  🤜 🤛